src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -23,10 +26,69 @@
 
     computed: {
       ...mapState('flow', ['flows']),
+
+      canvasHeight() {
+        const defaultHeight = window.screen.availHeight
+        if (!this.flows[0].blocks && this.flows[0].blocks.length) {
+          return defaultHeight
+        }
+
+        const blockAtTheLowestPosition = lodash.maxBy(this.flows[0].blocks, 'platform_metadata.io_viamo.uiData.yPosition')
+
+        if (!blockAtTheLowestPosition) {
+          return defaultHeight
+        }
+
+        const blockElement = document.getElementById(`block/${blockAtTheLowestPosition.uuid}`)
+
+        if (!blockElement) {
+          return defaultHeight
+        }
+
+        const blockHeight = blockElement.clientHeight
+        return lodash.get(blockAtTheLowestPosition, 'platform_metadata.io_viamo.uiData.yPosition') + blockHeight + 100
+      },
+
+      canvasWidth() {
+        const defaultWidth = window.screen.availWidth
+        if (!this.flows[0].blocks && this.flows[0].blocks.length) {
+          return defaultWidth
+        }
+
+        const blockAtTheFurthestRightPosition = lodash.maxBy(this.flows[0].blocks, 'platform_metadata.io_viamo.uiData.xPosition')
+
+        if (!blockAtTheFurthestRightPosition) {
+          return defaultWidth
+        }
+
+        const blockElement = document.getElementById(`block/${blockAtTheFurthestRightPosition.uuid}`)
+
+        if (!blockElement) {
+          return defaultWidth
+        }
+
+        const blockWidth = blockElement.clientWidth
+        return lodash.get(blockAtTheFurthestRightPosition, 'platform_metadata.io_viamo.uiData.xPosition') + blockWidth + 100
+      },
     },
 
     mounted() {
       this.$store.dispatch('builder/loadFlow')
+
+      // Auto scroll, needed to put in mounted()
+      this.$watch('canvasHeight', (newValue) => {
+        window.scrollTo({
+          top: newValue,
+          behavior: 'smooth'
+        })
+      }, { immediate: true });
+
+      this.$watch('canvasWidth', (newValue) => {
+        window.scrollTo({
+          left: newValue,
+          behavior: 'smooth'
```

Comment:
Nice one @safydy! This is such a nice + simple solution! 

Can you help me understand a handful of things?
1. Does this then scroll the entire page over by the entire width (for eg) as soon as I touch the edge? Would you be willing to attach a gif or vid of this in action on the PR description? Thanks!
2. Can we debounce/throttle the `scrollTo()` invocations so they're not called too rapidly?
3. Can you help me understand what influenced the decision to set up watchers in `mounted()` rather than as a top-level definition as per: https://vuejs.org/v2/guide/computed.html#Watchers ?
4. Is `{ immediate: true }` absolutely necessary here?

Reply:
@bzabos : 
Thanks 😊 ,
- for 1 & 2: I have thought about adding debounce when implementing, but it seems not working correctly. Here are 03 videos about 03 solutions : scroll smooth, scroll auto (not smooth), scroll auto with debounce (tried many delays) [https://drive.google.com/drive/folders/10oyAphd1mLK60kthVqQoLH6p2ji5IQdv?usp=sharing](https://drive.google.com/drive/folders/10oyAphd1mLK60kthVqQoLH6p2ji5IQdv?usp=sharing) 

> I opted for the **scroll smooth** solution

- for 3: outside mounted, I had an unavailable prop. But just check it and found I made a mistake on certain condition to return default value. Not it's working, and we don't need to put it inside mounted. Thanks for insisting here.
- for 4. I think it's not necessary, removed it.

Reply:
@bzabos is going to leave an example of how we could use `debounce()` to support this feature.

Reply:
To debounce a method: we need to cache a reference to the debounced version, then invoke that debounced version. I think this example could help: 
```
export default {
  methods: {
    debounceMethod: _.debounce(() => {
      console.log('Debounce button clicked!')
    }, 2000)
  }
}
```

via: https://www.digitalocean.com/community/tutorials/vuejs-lodash-throttle-debounce#debouncing-methods

Reply:
I used debounce for now to give us more control on the `timer`